datafusion/core/src/execution/runtime_env.rs

```diff
@@ -26,9 +26,11 @@ use std::collections::HashMap;
 
 use crate::datasource::datasource::TableProviderFactory;
 use crate::datasource::listing_table_factory::ListingTableFactory;
-use crate::datasource::object_store::ObjectStoreRegistry;
```

Comment:
the point of this PR is to remove the dependency here on `datasource`

The other references to datasource are removed in https://github.com/apache/arrow-datafusion/pull/5477